wallet/chainntfns.go

```diff
@@ -44,7 +44,7 @@ func (w *Wallet) extendMainChain(op errors.Op, dbtx walletdb.ReadWriteTx, header
 		return nil, errors.E(op, err)
 	}
 
-	var watch []wire.OutPoint
+	watch := make([]wire.OutPoint, 0, len(transactions))
```

Comment:
this isn't really an optimization. the number of outpoints it appends is not equal to the number of transactions.

Reply:
reverted